### PR TITLE
Delete unused InteropServices types

### PR DESCRIPTION
**Pull request by jamesqo**
Part of https://github.com/dotnet/coreclr/issues/4651

/cc @jkotas

Note: I will be focusing on removing only the unused types that take up significant LOC after this PR. Later, I will make another pass to eliminate the small fish such as individual members and tiny types like attribute definitions/enums.

### CODE REVIEW

**Comment by jkotas** [comment]
There are build errors - `error CS0246: The type or namespace name 'ObjectCreationDelegate' could not be found`

Looks good to me otherwise.

cc @yizhang82

**Comment by yizhang82** [comment]
@jamesqo Thanks for creating this PR. Can you help me understand what's the motivation of this change? This removal seems to be conflicting with a lot of netstandard 2.0 additions. For example, IRegistrationServices/IReflect/TypeLibFuncAttribute are definitely there in netstandard 2.0.

**Comment by jamesqo** [comment]
@yizhang82, @jkotas suggested in https://github.com/dotnet/coreclr/issues/4651#issuecomment-269583315 that I could remove types/methods that are being removed by BclRewriter as part of cleanup. I made a program to print out a list of types and members that are being removed ([gist](https://gist.github.com/jamesqo/2b848c6fec404726ac7bee1faae94324), [raw list](https://gist.githubusercontent.com/jamesqo/2b848c6fec404726ac7bee1faae94324/raw/2aaa3b5eddffe3ca224d49b78421fcafec5495e6/RemovableTypesAndMembers.txt)). The list includes IRegistrationServices/IReflect/TypeLibFuncAttribute, so if they are there, they are being removed by the rewriter.

**Comment by jkotas** [comment]
> IRegistrationServices, IReflect, TypeLibFuncAttribute

None of these are in NS2.0. Note that there are multiple IReflect interfaces. The one removed here is not the one in NS2.0.

**Comment by jkotas** [comment]
> Can you help me understand what's the motivation of this change? 

The motivation for this change is to delete sources that are not getting compiled into shipping corelib; and get rid of the BCL rewriter eventually.

File: src/mscorlib/src/System/Runtime/InteropServices/ITypeLibConverter.cs [hunk 1]
```diff
@@ -22,40 +22,6 @@ namespace System.Runtime.InteropServices {
...
-        None                            = 0x00000000,
-        OnlyReferenceRegistered         = 0x00000001,
-        CallerResolvedReferences        = 0x00000002,
-        OldNames                        = 0x00000004,
-        ExportAs32Bit                   = 0x00000010,
-        ExportAs64Bit                   = 0x00000020,        
-    }
-
     [Serializable]
 [System.Runtime.InteropServices.ComVisible(true)]
```
Comment (jkotas) [inline src/mscorlib/src/System/Runtime/InteropServices/ITypeLibConverter.cs]: Why are these two enums staying?

**Comment by yizhang82** [comment]
@jkotas Sorry, IReflect is probably bad example - that one is not the one we use in the runtime. However we do have IRegistrationServices/TypeLibFuncAttribute in the interop extension (not in "core" netstandard 2.0, you are right). What's our policy for the extensions? My understanding is that they are platform/runtime specific extensions that we can choose to implement, as extensions to netstandard 2.0 if people want those functionalities and/or for compact. Examples are TypeLib related attributes that I want to keep to be compatible with interop assemblies generated from TlbImp. (Technically they won't necessarily cause TypeLoadExceptions, but it's still nice to keep them for consistency and round-tripping, as well as custom authored interop assemblies in user C# code)
cc @weshaggard

**Comment by jkotas** [comment]
If we choose to add these as extension, we can bring them back. I would expect that a lot of these would not live corelib. For example, RegistrationServices just writes a bunch of registry entries - it does not need to be in corelib.

**Pull request merged**